spec/e2e/transaction_spec.rb

```diff
@@ -23,4 +23,37 @@
       expect(a.thing).to eq(b)
     end
   end
+
+  describe 'transaction behaviour when a validation fails' do
+    let(:clazz) do
+      UniqueClass.create do
+        include Neo4j::ActiveNode
+        property :name
+        validates :name, presence: true
+      end
+    end
+
+    it 'rollbacks the current transaction by default' do
+      expect do
+        Neo4j::Transaction.run do |tx|
+          clazz.create
+          expect(tx).to be_failed
+          clazz.create(name: 'john')
+        end
+      end.not_to change { clazz.count }
+    end
+
+    it 'keeps running the current transaction when `fail_transaction_when_validations_fail` is false' do
+      Neo4j::Config[:fail_transaction_when_validations_fail] = false
```

Comment:
There's a `let_config` helper in the `spec_helper` to temporarily define configuration values.  Definition:

https://github.com/neo4jrb/neo4j/blob/master/spec/spec_helper.rb#L116

Example:

https://github.com/neo4jrb/neo4j/blob/master/spec/e2e/active_model_spec.rb#L207